library/qtest/src/Tests.qs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+function Main() : Unit {
+    Std.Diagnostics.Fact(
+        Functions.TestCases([
+            ("Should return 42", TestCaseOne, 42),
+            ("Should add one", () -> AddOne(5), 6)
+        ]),
+        "Test harness failed to return true for all passing tests."
+    );
+    Std.Diagnostics.Fact(
+        Length(Functions.TestCasesSilent([
+            ("Should return 42", TestCaseOne, 43),
+            ("Should add one", () -> AddOne(5), 42)
+        ])) == 2,
+        "Test harness failed to return messages for failing tests."
+    );
```

Comment:
Could be good to assert that the returned messages are as expected, in addition to asserting there are the correct number of them. Would also be good to have a test case that mixes in some passing test cases with some failing test cases.

Reply:
Added some more tests here including the ones you outlined :)